src/main/java/frc/trigon/robot/subsystems/swerve/Swerve.java

```diff
@@ -261,36 +258,75 @@ void selfRelativeDrive(double xPower, double yPower, MirrorableRotation2d target
         selfRelativeDrive(speeds);
     }
 
-    private void selfRelativeDrive(ChassisSpeeds chassisSpeeds) {
-        chassisSpeeds = discretize(chassisSpeeds);
-        if (isStill(chassisSpeeds)) {
+    /**
+     * This method will take in desired robot-relative chassis targetSpeeds,
+     * generate a swerve setpoint, then set the target state for each module
+     *
+     * @param targetSpeeds The desired robot-relative targetSpeeds
+     */
+    private void selfRelativeDrive(ChassisSpeeds targetSpeeds) {
+        previousSetpoint = setpointGenerator.generateSetpoint(
+                previousSetpoint,
+                targetSpeeds,
+                RobotHardwareStats.getPeriodicTimeSeconds()
```

Comment:
use the actual passed period by like the discretize logic

Reply:
Why? The pp docs say to use the loop time

Reply:
more accurate and optimized